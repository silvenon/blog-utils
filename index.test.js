const { extractPostMetadata } = require('./')
const { getFileMd, getFileMdx } = require('./test/utils')

describe('extractPostMetadata', () => {
  it('takes a file path', async () => {
    await expect(
      extractPostMetadata(getFileMd().path, {
        frontmatter: true,
      }),
    ).resolves.toHaveProperty('frontmatter')
  })

  it('takes a vfile', async () => {
    await expect(
      extractPostMetadata(getFileMd(), {
        frontmatter: true,
      }),
    ).resolves.toHaveProperty('frontmatter')
  })

  it('supports MDX', async () => {
    await expect(
      extractPostMetadata(getFileMdx(), {
        frontmatter: true,
        isMdx: true,
      }),
    ).resolves.toHaveProperty('frontmatter')
  })

  describe('extracts metadata', () => {
    describe.each`
      language      | isMdx
      ${'Markdown'} | ${false}
      ${'MDX'}      | ${true}
    `('for $language', ({ isMdx }) => {
      test('all', async () => {
        await expect(
          extractPostMetadata(getFileMd(), {
            frontmatter: true,
            dateFromPath: true,
            excerpt: true,
            isMdx,
          }),
        ).resolves.toMatchSnapshot()
      })

      test('only frontmatter', async () => {
        await expect(
          extractPostMetadata(getFileMd(), { frontmatter: true, isMdx }),
        ).resolves.toMatchSnapshot()
      })

      test('only date', async () => {
        await expect(
          extractPostMetadata(getFileMd(), { dateFromPath: true, isMdx }),
        ).resolves.toMatchSnapshot()
      })

      test('only excerpt', async () => {
        await expect(
          extractPostMetadata(getFileMd(), { excerpt: true, isMdx }),
        ).resolves.toMatchSnapshot()
      })
    })
  })

  it.todo('supports other frontmatter formats')
})
