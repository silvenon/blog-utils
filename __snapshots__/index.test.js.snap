// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`extractPostMetadata extracts metadata for MDX all 1`] = `
Object {
  "date": "1859-11-24",
  "excerpt": "<blockquote>
<p>We all descended from monkeys, yo.</p>
<p>--Charles Darwin</p>
</blockquote>
",
  "frontmatter": Object {
    "author": "Charles Darwin",
    "title": "On the Origin of Species",
  },
}
`;

exports[`extractPostMetadata extracts metadata for MDX only date 1`] = `
Object {
  "date": "1859-11-24",
}
`;

exports[`extractPostMetadata extracts metadata for MDX only excerpt 1`] = `
Object {
  "excerpt": "<blockquote>
<p>We all descended from monkeys, yo.</p>
<p>--Charles Darwin</p>
</blockquote>
",
}
`;

exports[`extractPostMetadata extracts metadata for MDX only frontmatter 1`] = `
Object {
  "frontmatter": Object {
    "author": "Charles Darwin",
    "title": "On the Origin of Species",
  },
}
`;

exports[`extractPostMetadata extracts metadata for Markdown all 1`] = `
Object {
  "date": "1859-11-24",
  "excerpt": "<blockquote>
<p>We all descended from monkeys, yo.</p>
<p>--Charles Darwin</p>
</blockquote>
",
  "frontmatter": Object {
    "author": "Charles Darwin",
    "title": "On the Origin of Species",
  },
}
`;

exports[`extractPostMetadata extracts metadata for Markdown only date 1`] = `
Object {
  "date": "1859-11-24",
}
`;

exports[`extractPostMetadata extracts metadata for Markdown only excerpt 1`] = `
Object {
  "excerpt": "<blockquote>
<p>We all descended from monkeys, yo.</p>
<p>--Charles Darwin</p>
</blockquote>
",
}
`;

exports[`extractPostMetadata extracts metadata for Markdown only frontmatter 1`] = `
Object {
  "frontmatter": Object {
    "author": "Charles Darwin",
    "title": "On the Origin of Species",
  },
}
`;
